tracing-tracy/src/lib.rs

```diff
@@ -80,19 +86,25 @@ impl Default for TracyLayer {
     }
 }
 
-impl<S> Layer<S> for TracyLayer
+impl<S, F> Layer<S> for TracyLayer<F>
 where
     S: Subscriber + for<'a> registry::LookupSpan<'a>,
+    F: for<'writer> FormatFields<'writer> + 'static,
 {
     fn on_enter(&self, id: &Id, ctx: Context<S>) {
         if let Some(span_data) = ctx.span(id) {
             let metadata = span_data.metadata();
             let file = metadata.file().unwrap_or("<error: not available>");
             let line = metadata.line().unwrap_or(0);
+            let name = if let Some(fields) = span_data.extensions().get::<FormattedFields<F>>() {
+                format!("{}: {}", metadata.name(), fields.fields.as_str())
+            } else {
+                metadata.name().to_string()
```

Comment:
Consider using a `Cow<'_, str>` here, perhaps? It would also make sense to me to lift and duplicate the `Span::new` call into the conditional here.